django_sharding_library/router.py

```diff
@@ -22,10 +22,16 @@ def get_specific_database_or_none(self, model):
         return getattr(model, 'django_sharding__database', None)
 
     def get_shard_for_instance(self, instance):
-        return instance._state.db or instance.get_shard()
+        try:
+            return instance._state.db or instance.get_shard()
+        except:
+            return None
 
     def get_shard_for_id_field(self, model, sharded_by_field_id):
-        return model.get_shard_from_id(sharded_by_field_id)
+        try:
+            return model.get_shard_from_id(sharded_by_field_id)
+        except:
```

Comment:
Actually the old code just excepted and then tried the same thing again (not sure why) so I just removed the badly handled exception.